users/user-guide-query/multi-stage-query/operator-types/sortOrLimit.md

```diff
@@ -0,0 +1,183 @@
+is an inverted index in the `deviceOS` column.
+
+But if we modify the query to add a `limit` to the `userAttributes` table scan:
+```sql
+select 
+a.* 
+from (select * from userAttributes limit 10) as a
+join userGroups as g
+on a.userUUID = g.userUUID
+where a.deviceOS = 'windows'
+```
+
+The generated plan will be:
+```
+LogicalProject(daysSinceFirstTrip=[$0], deviceOS=[$1], totalTrips=[$2], userUUID=[$3])
+  LogicalJoin(condition=[=($3, $4)], joinType=[inner])
+    PinotLogicalExchange(distribution=[hash[3]])
+      LogicalFilter(condition=[=($1, _UTF-8'windows')])
+        LogicalSort(offset=[0], fetch=[10])
+          PinotLogicalSortExchange(distribution=[hash], collation=[[]], isSortOnSender=[false], isSortOnReceiver=[false])
+            LogicalSort(fetch=[10])
+              LogicalProject(daysSinceFirstTrip=[$3], deviceOS=[$4], totalTrips=[$5], userUUID=[$6])
+                LogicalTableScan(table=[[default, userAttributes]])
+    PinotLogicalExchange(distribution=[hash[0]])
+      LogicalProject(userUUID=[$4])
+        LogicalTableScan(table=[[default, userGroups]])
+```
+
+Here we can see that the filter `deviceOS = 'windows'` is not pushed down leaf stage, which
+means that the engine will need to scan all the data in the `userAttributes` table and then apply the filter.
+
+The reason why the filter is not pushed down is that the `limit` operation must be applied before the filter in order
+to not break the semantics, which in this case are saying that we want 10 rows of the `userAttributes` table
+without considering their `deviceOS` value.
+
+In cases where you actually want to apply the filter before the `limit`, you can specify the where clause in the 
+subquery. For example:
+
+```sql
+select 
+a.* 
+from (select * from userAttributes where deviceOS = 'windows' limit 10) as a
+join userGroups as g
+on a.userUUID = g.userUUID
+```
+
+Which will produce the following plan:
+```
+LogicalProject(daysSinceFirstTrip=[$0], deviceOS=[$1], totalTrips=[$2], userUUID=[$3])
+  LogicalJoin(condition=[=($3, $4)], joinType=[inner])
+    PinotLogicalExchange(distribution=[hash[3]])
+      LogicalSort(offset=[0], fetch=[10])
+        PinotLogicalSortExchange(distribution=[hash], collation=[[]], isSortOnSender=[false], isSortOnReceiver=[false])
+          LogicalSort(fetch=[10])
+            LogicalProject(daysSinceFirstTrip=[$3], deviceOS=[$4], totalTrips=[$5], userUUID=[$6])
+              LogicalFilter(condition=[=($4, _UTF-8'windows')])
+                LogicalTableScan(table=[[default, userAttributes]])
+    PinotLogicalExchange(distribution=[hash[0]])
+      LogicalProject(userUUID=[$4])
+        LogicalTableScan(table=[[default, userGroups]])
+```
+
+As you can see, the filter is pushed down to leaf stage, which will reduce the amount of data
+
+### Do not abuse `offset` pagination
+Although `OFFSET` and `LIMIT` are a very simple way to paginate results, they can be very inefficient.
```

Comment:
link to our pagination docs (or GH issue) here, so folks know an option exists / is coming soon?